src/polyhedralGravity/model/Polyhedron.cpp

```diff
@@ -225,27 +223,27 @@ namespace polyhedralGravity {
         const Array3 h = cross(rayVector, edge2);
         const double a = dot(edge1, h);
         if (a > -EPSILON_ZERO_OFFSET && a < EPSILON_ZERO_OFFSET) {
-            return nullptr;
+            return {};
         }
 
         const double f = 1.0 / a;
         const Array3 s = rayOrigin - triangle[0];
         const double u = f * dot(s, h);
         if (u < 0.0 || u > 1.0) {
```

Comment:
I want to remove the intersection logic from the Polyhedron and move it to the LeafNode since the tree will be responsible for intersection calculation in the future. Since Polyhedron is in the forked repo I wanted to communicate this change before putting it into action.